src/web/core/src/main/java/org/geoserver/web/data/resource/BasicResourceConfig.java

```diff
@@ -105,13 +93,14 @@ public BasicResourceConfig(String id, IModel<ResourceInfo> model) {
         refForm.add(latLonPanel);
         refForm.add(computeLatLonBoundsLink(refForm, nativeBBox, latLonPanel));
 
-        // check other native srs for Cascaded WFS or WMS resource
-        List<String> otherSRS = getOtherSRS(model.getObject());
-
-        // native srs , declared srs, and srs handling dropdown
+        // ALERT - CORE CHANGE
+        // for resources coming from WMS and WFS Datastore
+        // then check if it has more than one SRS, then add FIND SRS button in Native SRS
+        // which will allow user to set other SRS as Native SRS
+        List<String> otherSRS = getOtherSRSFromWFSOrWMS(model.getObject());
         CRSPanel nativeCRS;
-
         if (otherSRS.isEmpty()) {
```

Comment:
IMHO this should be a check validating if this is a remote cascaded WFS or WMS store, the inside if that's the case you may try to retrieve multiple SRS, makes sense?

Reply:
the method `getOtherSRSFromWFSOrWMS `only works on WFS and WMS Stores, for any other type of resource the list will come **empty**. I will add this to the comments